docs(open-source): EX-4666 GitHub specific documentation (#136)
empathyco/x: .github/contributing/commits.md

# Commits

All commits must follow the
[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification syntax.

> The `CHANGELOG.md` file is generated automatically using standard version that requires the commits to be in a specific format in order to be parsed.

Reviewer: Might be a good idea to add the link to `standard-version` plugin.

```suggestion
The `CHANGELOG.md` file is generated automatically using [standard-verison](https://www.npmjs.com/package/standard-version), which requires the commits to be in a specific format in order to be parsed.
```

- [ ] Add link to `standard-verison`

### Type

The commit type part must be one defined by the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)

- **build**: Changes that affect the build system or external dependencies.
- **ci**: Changes to our CI configuration files and scripts.

> - **chore**: Changes that you don’t want to appear in the change log. For example, this can be used to fix the change log if any release goes wrong.

Reviewer: ```suggestion
- **chore**: Changes that you don’t want to appear in the changelog. For example, this can be used to fix the changelog if any release goes wrong.
```

- [ ] Fix `change log` typos

> - **chore**: Changes that you don’t want to appear in the change log. For example, this can be used to fix the change log if any release goes wrong.

Reviewer: Is it `change-log`, `change log`, or `changelog`? I usually see it as just one word.

Reviewer 2: Agree

Author: Linguistically speaking it should be change log (as it's a log of changes), but [GitHub has gone for changelog](https://github.com/conventional-changelog/standard-version/issues/299). Example: https://docs.github.com/en/graphql/overview/changelog

I'll unify throughout.

- **docs**: Documentation only changes.
- **feat**: A new feature. New components, services, utilities…

> - **fix**: A bug fix, or a hot fix

Reviewer: `hot fix` should be just a word `hotfix`. `bug fix` can be either though 🤷‍♂️ .

```suggestion
- **fix**: A bug fix, or a hotfix
```

- [ ] change to `hotfix`

- **perf**: A code change that improves performance.
- **refactor**: An internal code change that neither fixes a bug nor adds a feature while keeping the same public API of the affected parts.
- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
- **test**: Adding missing tests or correcting existing tests. Every new feature must have its own test, so if you are adding a new component with its tests, the type should be **feat**, not **test**.

### Optional scope

The scope is optional. If a change affects the whole project, i.e. changing Prettier rules, you do not need to add a scope that says “all the project”. This part must only be added if the change affects a specific part of the code.

> The scope is used to group the related changes in the change log, so you can easily see all the changes that have been done to an specific part of the library.

Reviewer: ```suggestion
The scope is used to group the related changes in the changelog, so you can easily see all the changes that have been done to a specific part of the library.
```

- [ ] Fix `change log` and `an specific` typos

  - **related-tags**
  - **search**
  - **search-box**
- **x-plugin**: For changes that affect the core of the `x-components` project. For example, if you modify the X Components plugin, the X Components archetype, or the X bus.
- **wiring**: Changes that only affect the wiring operators or wiring factory.
- **e2e**: For end-to-end tests.
- **design-system**: For design tokens, CSS styling, etc.

### Description

The description is a short sentence written in **present tense** with an **imperative mood**. The description must explain the intention of the change at a high level. You don’t have to give details on how or why the change has been done; just tell what the change is about.

> added `Dropdown` component ❌ The message uses past tense

> add `Dropdown` component so it can be reused in the `SortDropdown` component that will be done in
> another PR ❌ Too specific

> adds `Dropdown` component ❌ Does not use imperative mood

> add `Dropdown` component ✅ Just perfect!

### Optional body

> If a simple description is not enough to describe the change, you give more details in the body of the commit. The body is optional. Remember that the purpose of using this commit format is to automate change log generation.

Reviewer: ```suggestion
If a simple description is not enough to describe the change, you give more details in the body of the commit. The body is optional. Remember that the purpose of using this commit format is to automate changelog generation.
```

- [ ] Fix `change log typo`

> This change log is read by all users of the library, so keep the level of details low.

Reviewer: ```suggestion
This changelog is read by all users of the library, so keep the level of details low.
```

- [ ] Fix `change log` typo

Users want to know how upgrading a version will affect their project, but they don’t care about internal or private changes.

```

> The `Dropdown` component serves as a replacement of the standard HTML `select` component. It exposes the same API to set the values and selected values. In addition, it also allows you to configure the animation to open and close the dropdown, and determine how each item is rendered.

Reviewer: ```suggestion
The `Dropdown` component serves as a replacement for the standard HTML `select` component. It exposes the same API to set the values and selected values. In addition, it also allows you to configure the animation to open and close the dropdown, and determine how each item is rendered.
```

- [ ] Fix `of` typo

#### Issue

> To properly link the commit with the issue ID, add it to the end of the commit footer. The change log includes a link to the task so users can read more about the motivation of the change.

Reviewer: ```suggestion
To properly link the commit with the issue ID, add it to the end of the commit footer. The changelog includes a link to the task so users can read more about the motivation of the change.
```

- [ ] Fix `change log` typo
